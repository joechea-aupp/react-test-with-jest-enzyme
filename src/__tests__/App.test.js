import React from "react";
import { mount } from "enzyme";
import { act } from "react-dom/test-utils";
import App from "../App";

jest.mock("../services", () => {
  return {
    getToDos: () =>
      Promise.resolve([
        {
          title: "Book concert tickets",
          done: false,
          id: 1
        },
        {
          title: "Buy a new iPhone",
          done: true,
          id: 2
        },
        {
          title: "Write an article on React",
          done: true,
          id: 3
        }
      ]),
    addTodo: obj => Promise.resolve({ ...obj, id: 4 })
  };
});

describe("Testing StatusButton component", () => {
  it("Renders a snapshot successfully!", async () => {
    let wrapper;
    await act(async () => {
      wrapper = mount(<App />);
    });

    expect(wrapper.html()).toMatchSnapshot();

    wrapper
      .find(".add-todo > input")
      .simulate("change", {
        target: { value: "This is an amazing task!" }
      })
      .simulate("keyup", { key: "Enter", keyCode: 13 });

    await act(async () => wrapper.update());

    expect(wrapper.html()).toMatchSnapshot();
  });
});
