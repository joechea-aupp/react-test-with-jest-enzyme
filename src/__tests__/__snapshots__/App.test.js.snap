// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing StatusButton component Renders a snapshot successfully! 1`] = `"<div class=\\"app\\"><div class=\\"title\\">Tasks</div><div class=\\"add-todo\\"><input type=\\"text\\" placeholder=\\"Add a task\\" value=\\"\\"></div><div class=\\"list\\"><div class=\\"list-item\\"><div class=\\"task\\">Book concert tickets</div><div class=\\"status-btn\\"></div><div class=\\"delete-btn\\">X</div></div><div class=\\"list-item\\"><div class=\\"task\\">Buy a new iPhone</div><div class=\\"status-btn status-done\\"></div><div class=\\"delete-btn\\">X</div></div><div class=\\"list-item\\"><div class=\\"task\\">Write an article on React</div><div class=\\"status-btn status-done\\"></div><div class=\\"delete-btn\\">X</div></div></div></div>"`;

exports[`Testing StatusButton component Renders a snapshot successfully! 2`] = `"<div class=\\"app\\"><div class=\\"title\\">Tasks</div><div class=\\"add-todo\\"><input type=\\"text\\" placeholder=\\"Add a task\\" value=\\"\\"></div><div class=\\"list\\"><div class=\\"list-item\\"><div class=\\"task\\">Book concert tickets</div><div class=\\"status-btn\\"></div><div class=\\"delete-btn\\">X</div></div><div class=\\"list-item\\"><div class=\\"task\\">Buy a new iPhone</div><div class=\\"status-btn status-done\\"></div><div class=\\"delete-btn\\">X</div></div><div class=\\"list-item\\"><div class=\\"task\\">Write an article on React</div><div class=\\"status-btn status-done\\"></div><div class=\\"delete-btn\\">X</div></div><div class=\\"list-item\\"><div class=\\"task\\">This is an amazing task!</div><div class=\\"status-btn\\"></div><div class=\\"delete-btn\\">X</div></div></div></div>"`;
